src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -227,6 +225,50 @@ public JsonClassInfo(Type type, JsonSerializerOptions options)
             }
         }
 
+        private void CacheMember(
+            Type declaringType,
+            Type memberType,
+            MemberInfo memberInfo,
+            Dictionary<string, JsonPropertyInfo> cache,
+            ref Dictionary<string, MemberInfo>? ignoredMembers)
+        {
+            JsonPropertyInfo jsonPropertyInfo = AddProperty(memberInfo, memberType, declaringType, Options);
+            Debug.Assert(jsonPropertyInfo.NameAsString != null);
+
+            string memberName = memberInfo.Name;
+
+            // The JsonPropertyNameAttribute or naming policy resulted in a collision.
```

Comment:
We've made a few changes recently to make sure we have the correct semantics for property name collisions. We should be sure to add tests to cover collisions between field names, and also between property and field names. Basically, we should extend this test suite: https://github.com/dotnet/runtime/blob/8fd53646fc1a099db797d0177b7d9d1eac581939/src/libraries/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs#L14-L431